internal/services/pam/pam_test.go

```diff
@@ -0,0 +1,595 @@
+				}
+			}
+			if tc.sessionID == "-" {
+				tc.sessionID = ""
+			}
+
+			esReq := &authd.ESRequest{
+				SessionId: tc.sessionID,
+			}
+			_, err := testClient.EndSession(context.Background(), esReq)
+			if tc.wantErr {
+				require.Error(t, err, "EndSession should return an error, but did not")
+				return
+			}
+			require.NoError(t, err, "EndSession should not return an error, but did")
+		})
+	}
+}
+
+func startClient() (client authd.PAMClient, cleanup func(), err error) {
+	tmpDir, err := os.MkdirTemp("", "authd-internal-pam-tests-")
+	if err != nil {
+		return nil, nil, err
+	}
+
+	brokerDir := filepath.Join(tmpDir, "etc", "authd", "broker.d")
+	if err = os.MkdirAll(brokerDir, 0750); err != nil {
+		_ = os.RemoveAll(tmpDir)
+		return nil, nil, err
+	}
+	_, brokerCleanup, err := testutils.StartBusBrokerMock(brokerDir, "BrokerMock")
+	if err != nil {
+		_ = os.RemoveAll(tmpDir)
+		return nil, nil, err
+	}
+
+	socketPath := filepath.Join(tmpDir, "authd.sock")
+	lis, err := net.Listen("unix", socketPath)
+	if err != nil {
+		_ = os.RemoveAll(tmpDir)
+		brokerCleanup()
+		return nil, nil, err
+	}
+	// We want everyone to be able to write to our socket and we will filter permissions
+	// #nosec G302
+	if err = os.Chmod(socketPath, 0666); err != nil {
+		_ = os.RemoveAll(tmpDir)
+		brokerCleanup()
+		return nil, nil, err
+	}
+
+	brokerManager, err := brokers.NewManager(context.Background(), nil, brokers.WithRootDir(tmpDir))
+	if err != nil {
+		_ = os.RemoveAll(tmpDir)
+		brokerCleanup()
+		return nil, nil, err
+	}
+
+	grpcServer := grpc.NewServer()
+	service := pam.NewService(context.Background(), brokerManager)
+	authd.RegisterPAMServer(grpcServer, service)
+	done := make(chan struct{})
+	go func() {
+		defer close(done)
+		_ = grpcServer.Serve(lis)
+	}()
+
+	conn, err := grpc.Dial("unix://"+socketPath, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	if err != nil {
+		grpcServer.Stop()
+		<-done
+		brokerCleanup()
+		_ = os.RemoveAll(tmpDir)
+		return nil, nil, err
+	}
+
+	return authd.NewPAMClient(conn), func() {
+		conn.Close()
+		grpcServer.Stop()
+		<-done
+		brokerCleanup()
+		_ = os.RemoveAll(tmpDir)
+	}, nil
+}
+
+func TestMain(m *testing.M) {
+	testutils.InstallUpdateFlag()
+	flag.Parse()
+
+	// Start system bus mock.
+	busCleanup, err := testutils.StartSystemBusMock()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "%v\n", err)
+		os.Exit(1)
+	}
+	defer busCleanup()
+
+	var cleanup func()
+	testClient, cleanup, err = startClient()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "%v\n", err)
+		busCleanup()
+		os.Exit(1)
+	}
+	defer cleanup()
+
+	brokerGeneratedID = getGeneratedID(testClient, "BrokerMock")
+	if brokerGeneratedID == "" {
+		fmt.Fprintf(os.Stderr, "could not get generated ID for BrokerMock\n")
+		cleanup()
+		busCleanup()
+		os.Exit(1)
+	}
+	m.Run()
+}
+
+// getGeneratedID returns the generated ID for the specified broker.
+func getGeneratedID(client authd.PAMClient, brokerName string) string {
```

Comment:
so, we renamed the variable to get clearer, I thought you would also update the associated function name to be clearer that this is from the broker.